app/services/mapping/goodsshipment/UCRBuilder.scala

```diff
@@ -26,22 +26,19 @@ object UCRBuilder {
   def build(implicit cacheMap: CacheMap): UCR =
     cacheMap
       .getEntry[ConsignmentReferences](ConsignmentReferences.id)
+      .filter(isDefined)
       .map(createUCR)
       .orNull
 
-  private def createUCR(data: ConsignmentReferences): UCR = {
+  private def isDefined(reference: ConsignmentReferences): Boolean = !reference.ducr.getOrElse(Ducr("")).ducr.isEmpty
```

Comment:
The same about `!` and `isEmpty`

I'm thinking about preparing isEmpty method inside Ducr class. Using will be easier and we can do it with every class that you're using for dec